### PR TITLE
Login via email address or username for OpenID

### PR DESCRIPTION
**Pull request by deanmarcussen**
Fixes https://github.com/OrchardCMS/OrchardCore/issues/10190

### CODE REVIEW

**Comment by kevinchalet** [comment]
IIRC, the login controller - that doesn't call this method - does additional checks like ensuring "local logins" are allowed. Is it something the user service should do too?

**Comment by deanmarcussen** [comment]
You’re probably right @kevinchalet I suspect it may not even check if the user is disabled.

**Comment by wAsnk** [comment]
Could this be merged? This bug still persists.

**Comment by Skrypt** [comment]
@deanmarcussen Is it still something that we should merge?

**Comment by deanmarcussen** [comment]
See comments from @kevinchalet - while this fixes one issue, it is not complete (at a glance)

**Comment by wAsnk** [comment]
> See comments from @kevinchalet - while this fixes one issue, it is not complete (at a glance)

While it is not complete it still fixes one bug.

**Comment by deanmarcussen** [comment]
You are welcome to fork it, and fix the remaining :)

File: src/OrchardCore/OrchardCore.Users.Core/Services/UserService.cs [hunk 1]
```diff
@@ -54,7 +54,7 @@ public async Task<IUser> AuthenticateAsync(string userName, string password, Act
                 return null;
             }
 
-            var user = await _userManager.FindByNameAsync(userName);
+            var user = await _userManager.FindByNameAsync(userName) ?? await _userManager.FindByEmailAsync(userName);
```
Comment (Piedone) [inline src/OrchardCore/OrchardCore.Users.Core/Services/UserService.cs:57]: Include this logic in `GetUserAsync()` instead and then:

```suggestion
            var user = await GetUserAsync(userName);
```

**Review by Piedone — CHANGES_REQUESTED** [review]

**Pull request closed without merging**